tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,662 @@
+    """Adds a repository to an existing content view.
+
+    :param int repo_id: The ID for an existing repository.
+    :param int cv_id: The ID for an existing content view.
+
+    """
+    client.put(
+        entities.ContentView(id=cv_id).path(),
+        auth=get_server_credentials(),
+        verify=False,
+        data={u'repository_ids': [repo_id]}
+    ).raise_for_status()
+
+
+def _add_content_view_to_composite_view(cv_id, cv_version_id):
+    """Adds a published content view to a composite content view.
+
+    :param int cv_id: The ID for an existing composite content view.
+    :param int cv_version_id: The ID for a published non-composite
+    content view.
+
+    """
+    client.put(
+        entities.ContentView(id=cv_id).path(),
+        auth=get_server_credentials(),
+        verify=False,
+        data={u'component_ids': [cv_version_id]}
+    ).raise_for_status()
+
+
+def _publish_content_view(cv_id):
+    """Publishes an existing content view.
+
+    :param int cv_id: The ID for an existing content view.
+
+    :return: A boolean indicating whether the publication worked or not.
+
+    """
+    task_status = entities.ContentView(id=cv_id).publish()
+
+    return task_status['result'] == u'success'
+
+
+@ddt
+class DockerRepositoryTestCase(APITestCase):
+    """Tests specific to using ``Docker`` repositories."""
+
+    @classmethod
+    def setUpClass(cls):  # noqa
+        """Create an organization and product which can be re-used in tests."""
+        cls.org_id = entities.Organization().create()['id']
+
+    @run_only_on('sat')
+    @data(
+        gen_string('alpha', 15),
+        gen_string('alphanumeric', 15),
+        gen_string('numeric', 15),
+        gen_string('latin1', 15),
+        gen_string('utf8', 15),
+        gen_string('html', 15),
+    )
+    def test_create_one_docker_repo(self, name):
+        """@Test: Create one Docker-type repository
+
+        @Assert: A repository is created with a Docker image.
+
+        @Feature: Docker
+
+        """
+        upstream_name = u'busybox'
+        content_type = u'docker'
+        prod_id = entities.Product(organization=self.org_id).create()['id']
+
+        repository = _create_repository(prod_id, name, None)
+        repo_id = repository['id']
+        real_attrs = entities.Repository(id=repo_id).read_json()
+        self.assertEqual(real_attrs['name'], name)
+        self.assertEqual(real_attrs['docker_upstream_name'], upstream_name)
+        self.assertEqual(real_attrs['content_type'], content_type)
+
+    @run_only_on('sat')
+    def test_create_multiple_docker_repo(self):
+        """@Test: Create multiple Docker-type repositories
+
+        @Assert: Multiple docker repositories are created with a Docker image
+        and they all belong to the same product.
+
+        @Feature: Docker
+
+        """
+        prod_id = entities.Product(organization=self.org_id).create()['id']
+
+        for _ in range(0, randint(1, 5)):
```

Comment:
The first argument to `range` can be omitted.

    >>> range(0, 3) == range(3)
    True